homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -350,6 +361,19 @@ async def async_migrate_entry(
 
         hass.config_entries.async_update_entry(entry, minor_version=2)
 
+    if entry.version == 2 and entry.minor_version == 2:
```

Comment:
I think this needs to be `entry.minor_version < 3`

Again that's what at least I do in #147563

Reply:
the previous step has updated it to minor_version=2